src/measure_theory/function/uniform_integrable.lean

```diff
@@ -636,4 +640,199 @@ lemma tendsto_in_measure_iff_tendsto_Lp [is_finite_measure μ] (hp : 1 ≤ p) (h
+  by_cases hμs' : μ s = 0,
+  { rw (snorm_eq_zero_iff ((hf i).indicator hs).ae_measurable hpzero).2 (indicator_meas_zero hμs'),
+    norm_num },
+  calc snorm (indicator s (f i)) p μ ≤ snorm (indicator (s ∩ {x | C ≤ ∥f i x∥₊}) (f i)) p μ +
+    snorm (indicator (s ∩ {x | ∥f i x∥₊ < C}) (f i)) p μ :
+    begin
+      refine le_trans (eq.le _) (snorm_add_le (measurable.ae_measurable
+        ((hf i).indicator (hs.inter (measurable_set_le measurable_const (hf i).nnnorm))))
+        (measurable.ae_measurable ((hf i).indicator (hs.inter
+        (measurable_set_lt (hf i).nnnorm measurable_const)))) hp),
+      congr,
+      change _ = λ x, (s ∩ {x : α | C ≤ ∥f i x∥₊}).indicator (f i) x +
+        (s ∩ {x : α | ∥f i x∥₊ < C}).indicator (f i) x,
+      rw ← set.indicator_union_of_disjoint,
+      { congr,
+        rw [← inter_union_distrib_left, (by { ext, simp [le_or_lt] } :
+          {x : α | C ≤ ∥f i x∥₊} ∪ {x : α | ∥f i x∥₊ < C} = set.univ), inter_univ] },
+      { refine (disjoint.inf_right' _ _).inf_left' _,
+        rintro x ⟨hx₁ : _ ≤ _, hx₂ : _ < _⟩,
+        exact false.elim (hx₂.ne (eq_of_le_of_not_lt hx₁ (not_lt.2 hx₂.le)).symm) }
+    end
+    ... ≤ snorm (indicator ({x | C ≤ ∥f i x∥₊}) (f i)) p μ + C * μ s ^ (1 / ennreal.to_real p) :
+    begin
+      refine add_le_add (snorm_mono $ λ x, norm_indicator_le_of_subset
+        (inter_subset_right _ _) _ _) _,
+      rw ← indicator_indicator,
+      rw snorm_indicator_eq_snorm_restrict,
+      have : ∀ᵐ x ∂(μ.restrict s), ∥({x : α | ∥f i x∥₊ < C}).indicator (f i) x∥ ≤ C,
+      { refine ae_of_all _ _,
+        simp_rw norm_indicator_eq_indicator_norm,
+        exact indicator_le' (λ x (hx : _ < _), hx.le) (λ _ _, nnreal.coe_nonneg _) },
+      refine le_trans (snorm_le_of_ae_bound this) _,
+      rw [mul_comm, measure.restrict_apply' hs, univ_inter,
+        ennreal.of_real_coe_nnreal, one_div],
+      exacts [le_rfl, hs],
+    end
+    ... ≤ ennreal.of_real (ε / 2) + C * ennreal.of_real (ε / (2 * C)) :
+    begin
+      refine add_le_add (hC i) (mul_le_mul_left' _ _),
+      rwa [ennreal.rpow_one_div_le_iff (ennreal.to_real_pos hpzero hp'),
+        ennreal.of_real_rpow_of_pos (div_pos hε (mul_pos two_pos (nnreal.coe_pos.2 hCpos)))]
+    end
+    ... ≤ ennreal.of_real (ε / 2) + ennreal.of_real (ε / 2) :
+    begin
+      refine add_le_add_left _ _,
+      rw [← ennreal.of_real_coe_nnreal, ← ennreal.of_real_mul (nnreal.coe_nonneg _),
+        ← div_div_eq_div_mul, mul_div_cancel' _ (nnreal.coe_pos.2 hCpos).ne.symm],
+      exact le_rfl,
+    end
+    ... ≤ ennreal.of_real ε :
+    begin
+      rw [← ennreal.of_real_add (half_pos hε).le (half_pos hε).le, add_halves],
+      exact le_rfl,
+    end
+end
+
+lemma unif_integrable_of (hp : 1 ≤ p) (hp' : p ≠ ∞) (hf : ∀ i, measurable (f i))
+  (h : ∀ ε : ℝ, 0 < ε → ∃ C : ℝ≥0,
+    ∀ i, snorm ({x | C ≤ ∥f i x∥₊}.indicator (f i)) p μ ≤ ennreal.of_real ε) :
+  unif_integrable f p μ :=
```

Comment:
Should this be placed in another section of the file? This is not about `uniform_integrable`.

Reply:
By "placed in another section", I meant that you should put it before the `uniform_integrable` section, in the already existing `unif_integrable` part, since it is about unif_integrable. The idea is to group together lemmas about similar objects.

Reply:
Done!